NullPointerException after system error message redirection

**Describe the bug**
qvisualise: test failed and report "NullPointerException" as below:
```
executeWithInvalidOutputFileType
java.lang.ExceptionInInitializerError
	at org.qcmg.qvisualise.QVisualise.setup(QVisualise.java:115)
	at org.qcmg.qvisualise.QVisualiseTest.executeWithInvalidOutputFileType(QVisualiseTest.java:137)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        ...
Caused by: java.lang.NullPointerException
	at java.util.logging.LogManager$5.run(LogManager.java:982)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.util.logging.LogManager.loadLoggerHandlers(LogManager.java:958)
	at java.util.logging.LogManager.initializeGlobalHandlers(LogManager.java:1578)
	at java.util.logging.LogManager.access$1500(LogManager.java:145)
	at java.util.logging.LogManager$RootLogger.accessCheckedHandlers(LogManager.java:1667)
	at java.util.logging.Logger.getHandlers(Logger.java:1777)
        ...
```

**Investigation**
I realize there is a test before sss, which redirect the error message to null at the last step. The Code is 
```
		ByteArrayOutputStream errContent = new ByteArrayOutputStream();
		System.setErr(new PrintStream(errContent));		
		String[] args = {"-h"};
		new QVisualise().setup(args);
		...
		System.setErr(null);
```
Java is cross-platform and '/dev/null' is Unix specific,  "null" may not support by other platforms. Here we should set it to the standard error output. Our org.qcmg.common.log.QLoggerFactory is confused after the redirection, hence it throws "NullPointerException".

**Solution**
 to set it back to original stderr.
```
		PrintStream origin = System.err;
		ByteArrayOutputStream errContent = new ByteArrayOutputStream();
		System.setErr(new PrintStream(errContent));		
		...
		System.setErr(origin);
```

--- a/qvisualise/src/org/qcmg/qvisualise/report/Report.java
+++ b/qvisualise/src/org/qcmg/qvisualise/report/Report.java
@@ -16,13 +16,10 @@ public class Report {
 	
 	private final ProfileType type;
 	private final String fileName;
-//	private final long recordParsed;
 	private final String recordParsed;
-//	private final long duplicates;
 	private final String duplicates;
 	private List<ChartTab> tabs;	
 	
-	//xu
 	private String runBy = null;
 	private String runOn = null;
 	private String version = null;
@@ -35,14 +32,6 @@ public class Report {
 		this.duplicates = duplicates;
 		tabs = new ArrayList<ChartTab>();
 	}
-	
-//	public Report(ProfileType type, String fileName, long records, long duplicates) {
-//	this.type = type;
-//	this.fileName = fileName;
-//	this.recordParsed = records;
-//	this.duplicates = duplicates;
-//	tabs = new ArrayList<ChartTab>();
-//}	
 	
 	
 	public void addTab(ChartTab tab) {
@@ -84,7 +73,6 @@ public class Report {
 	public String getVersion(){
 		return version;
 	}
-	
 	
 	public void setRunBy(String str){
 		 runBy = str;

--- a/qvisualise/src/org/qcmg/qvisualise/report/ReportBuilder.java
+++ b/qvisualise/src/org/qcmg/qvisualise/report/ReportBuilder.java
@@ -20,8 +20,6 @@ import java.util.TreeMap;
 import java.util.concurrent.atomic.AtomicLong;
 import java.util.concurrent.atomic.AtomicLongArray;
 
-import org.qcmg.common.log.QLogger;
-import org.qcmg.common.log.QLoggerFactory;
 import org.qcmg.common.model.CigarStringComparator;
 import org.qcmg.common.model.MAPQMiniMatrix;
 import org.qcmg.common.model.ProfileType;
@@ -40,7 +38,7 @@ import org.w3c.dom.NodeList;
 
 public class ReportBuilder {
 	
-	private static final QLogger logger = QLoggerFactory.getLogger(ReportBuilder.class);	
+//	private static final QLogger logger = QLoggerFactory.getLogger(ReportBuilder.class);	
 	private static final String ISIZE = "TLEN";
 	private static final String UNMAPPED = "Unmapped";
 	private static final String DUPLICATE = "Duplicate";

--- a/qvisualise/src/org/qcmg/qvisualise/report/ReportBuilder.java
+++ b/qvisualise/src/org/qcmg/qvisualise/report/ReportBuilder.java
@@ -24,7 +24,6 @@ import org.qcmg.common.model.CigarStringComparator;
 import org.qcmg.common.model.MAPQMiniMatrix;
 import org.qcmg.common.model.ProfileType;
 import org.qcmg.common.model.ReferenceNameComparator;
-//import org.qcmg.common.model.SummaryByCycle;
 import org.qcmg.qvisualise.ChartTab;
 import org.qcmg.qvisualise.Messages;
 import org.qcmg.qvisualise.QVisualiseException;
@@ -38,7 +37,6 @@ import org.w3c.dom.NodeList;
 
 public class ReportBuilder {
 	
-//	private static final QLogger logger = QLoggerFactory.getLogger(ReportBuilder.class);	
 	private static final String ISIZE = "TLEN";
 	private static final String UNMAPPED = "Unmapped";
 	private static final String DUPLICATE = "Duplicate";
@@ -285,7 +283,6 @@ public class ReportBuilder {
 						}
 					}					
 					
-					//*****start xu										
 					ChartTab ct = new ChartTab("Summary", "summ" + reportID);
 					String str = HTMLReportUtils.generateGoogleDataForTableStringMap(summaryMap, ct.getName()+1, "Property", "Value"  );	
 					
@@ -315,7 +312,6 @@ public class ReportBuilder {
 					List<ChartTab> chartTabs = report.getTabs();
 					chartTabs.add(0, ct);				
 					report.setTabs(chartTabs);
-					//*****end xu		
 				} else {
 					System.out.println("summaryNodes was null");
 				}

--- a/qvisualise/test/org/qcmg/qvisualise/QVisualiseTest.java
+++ b/qvisualise/test/org/qcmg/qvisualise/QVisualiseTest.java
@@ -1,12 +1,14 @@
 package org.qcmg.qvisualise;
 
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
 import java.io.ByteArrayOutputStream;
 import java.io.File;
 import java.io.FileWriter;
 import java.io.IOException;
 import java.io.PrintStream;
-
-import junit.framework.Assert;
 
 import org.junit.Ignore;
 import org.junit.Rule;
@@ -30,14 +32,14 @@ public class QVisualiseTest {
 		try {
 			QVisualise qv = new QVisualise();
 			qv.setup(args);
-			Assert.fail("Should have thrown an exception");
+			fail("Should have thrown an exception");
 		} catch (Exception e) {}
 		
 		String[] args2 = {"-input","file1.xml", "-i","ifile2.xml","-output","file2.xml", "-out","file3.xml"};
 		try {
 			QVisualise qv = new QVisualise();
 			qv.setup(args2);
-			Assert.fail("Should have thrown an exception");
+			fail("Should have thrown an exception");
 		} catch (Exception e) {}
 	}
 	
@@ -47,10 +49,10 @@ public class QVisualiseTest {
 		try {
 			QVisualise qv = new QVisualise();
 			int exitStatus = qv.setup(args);
-			Assert.assertEquals(1, exitStatus);
+			assertEquals(1, exitStatus);
 		} catch (Exception e) {
 			e.printStackTrace();
-			Assert.fail("no exception should have been thrown from executeWithNoArgs()");
+			fail("no exception should have been thrown from executeWithNoArgs()");
 		}
 	}
 	
@@ -63,9 +65,9 @@ public class QVisualiseTest {
 		String[] args = {"-input",inputFile.getAbsolutePath()};
 		try {
 			int exitStatus = new QVisualise().setup(args);
-			Assert.assertEquals(1, exitStatus);
+			assertEquals(1, exitStatus);
 		} catch (QVisualiseException qpe) {
-			Assert.fail("Should NOT have thrown a QVisualiseException");
+			fail("Should NOT have thrown a QVisualiseException");
 		}
 		
 		//fail due to missing content in input file
@@ -73,9 +75,9 @@ public class QVisualiseTest {
 		String[] args2 = {"-input",inputFile.getAbsolutePath(), "-log",logFile.getAbsolutePath()};
 		try {
 			int exitStatus = new QVisualise().setup(args2);
-			Assert.assertEquals(0, exitStatus);
+			assertEquals(0, exitStatus);
 		} catch (QVisualiseException qpe) {
-			Assert.fail("Should NOT have thrown a QVisualiseException");
+			fail("Should NOT have thrown a QVisualiseException");
 		}
 	}
 	
@@ -86,9 +88,9 @@ public class QVisualiseTest {
 		// will fail due to missing log file
 		try {
 			int exitStatus = new QVisualise().setup(args);
-			Assert.assertTrue(1 == exitStatus);
+			assertTrue(1 == exitStatus);
 		} catch (QVisualiseException qpe) {
-			Assert.fail("Should have thrown a QVisualiseException");
+			fail("Should have thrown a QVisualiseException");
 		}
 		
 		File inputFile = testFolder.newFile(TEST_XML);
@@ -100,18 +102,18 @@ public class QVisualiseTest {
 		// will fail due to empty file
 		try {
 			new QVisualise().setup(args2);
-			Assert.fail("Should have thrown a QVisualiseException");
+			fail("Should have thrown a QVisualiseException");
 		} catch (QVisualiseException qpe) {
-			Assert.assertEquals(true, qpe.getMessage().startsWith("Empty input file"));
+			assertEquals(true, qpe.getMessage().startsWith("Empty input file"));
 		}
 		
 		createSkeletonProfilerFile(inputFile);
 		// will succeed
 		try {
 			int exitStatus = new QVisualise().setup(args2);
-			Assert.assertTrue(0 == exitStatus);
+			assertTrue(0 == exitStatus);
 		} catch (QVisualiseException qpe) {
-			Assert.fail("Should not have thrown a QVisualiseException");
+			fail("Should not have thrown a QVisualiseException");
 		}
 	}
 	
@@ -121,9 +123,9 @@ public class QVisualiseTest {
 		String[] args = {"-input",TEST_TEST, "-log",logFile.getAbsolutePath()};
 		try {
 			new QVisualise().setup(args);
-			Assert.fail("Should have thrown a QVisualiseException");
+			fail("Should have thrown a QVisualiseException");
 		} catch (QVisualiseException qpe) {
-			Assert.assertTrue(qpe.getMessage().startsWith("Unsupported file type"));
+			 assertTrue(qpe.getMessage().startsWith("Unsupported file type"));
 		}
 	}
 	
@@ -133,28 +135,29 @@ public class QVisualiseTest {
 		String[] args = {"-input",TEST_XML, "-output",TEST_XML, "-log", logFile.getAbsolutePath()};
 		try {
 			new QVisualise().setup(args);
-			Assert.fail("Should have thrown a QVisualiseException");
+			fail("Should have thrown a QVisualiseException");
 		} catch (QVisualiseException qpe) {
-			Assert.assertTrue(qpe.getMessage().startsWith("Unsupported file type"));
+			assertTrue(qpe.getMessage().startsWith("Unsupported file type"));
 		}
 	}
 	
 	@Test
 	public final void executeWithOptions() throws Exception {
+		PrintStream origin = System.err;
 		ByteArrayOutputStream errContent = new ByteArrayOutputStream();
 		System.setErr(new PrintStream(errContent));
 		
 		String[] args = {"-h"};
 		new QVisualise().setup(args);
-		Assert.assertTrue(errContent.toString().contains(Messages.USAGE));
+		assertTrue(errContent.toString().contains(Messages.USAGE));
 		
 		errContent.reset();
 		
 		args[0] = "-v";
 		new QVisualise().setup(args);
-		Assert.assertTrue(errContent.toString().contains("null, version null"));
+		assertTrue(errContent.toString().contains("null, version null"));
 		
-		System.setErr(null);
+		System.setErr(origin);
 	}
 	
 	

